src/test/java/org/jenkinsci/plugins/gitclient/verifier/AcceptFirstConnectionVerifierTest.java

```diff
@@ -47,38 +68,77 @@ public void testVerifyServerHostKeyWhenFirstConnection() throws Exception {
+        // String hostKeyEntry =
+        //        "|1|I9eFW1PcZ6UvKPt6iHmYwTXTo54=|PyasyFX5Az4w9co6JTn7rHkeFII= ssh-ed25519
+        // AAAAC3NzaC1lZDI1NTE5AAAAIOMqqnkVzrm0SdG6UOoqKLsabgH5C9okWi0dh2l9GKJl";
         String hostKeyEntry =
-                "|1|I9eFW1PcZ6UvKPt6iHmYwTXTo54=|PyasyFX5Az4w9co6JTn7rHkeFII= ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAIOMqqnkVzrm0SdG6UOoqKLsabgH5C9okWi0dh2l9GKJl";
+                "|1|FJGXVAi7jMQIsl1J6uE6KnCiteM=|xlH92KQ91GuBgRxvRbU/sBo60Bo= ecdsa-sha2-nistp256 AAAAE2VjZHNhLXNoYTItbmlzdHAyNTYAAAAIbmlzdHAyNTYAAABBBEmKSENjQEezOmxkZMy7opKgwFB9nkt5YRrYMjNuG5N87uRgg6CLrbo5wAdT/y6v0mKV0U2w0WZ2YB/++Tpockg=";
+
         File mockedKnownHosts = knownHostsTestUtil.createFakeKnownHosts(hostKeyEntry);
         AcceptFirstConnectionVerifier acceptFirstConnectionVerifier = spy(new AcceptFirstConnectionVerifier());
         when(acceptFirstConnectionVerifier.getKnownHostsFile()).thenReturn(mockedKnownHosts);
-        AbstractJGitHostKeyVerifier verifier = acceptFirstConnectionVerifier.forJGit(TaskListener.NULL);
-        JGitConnection jGitConnection = new JGitConnection("github.com", 22);
 
-        // Should connect and do not add new line because keys are equal
-        jGitConnection.connect(verifier);
-        assertThat(mockedKnownHosts, is(anExistingFile()));
-        assertThat(Files.readAllLines(mockedKnownHosts.toPath()), is(Collections.singletonList(hostKeyEntry)));
+        KnownHostsTestUtil.connectToHost(
+                        "github.com",
+                        22,
+                        mockedKnownHosts,
+                        acceptFirstConnectionVerifier.forJGit(StreamBuildListener.fromStdout()),
+                        s -> {
+                            assertThat(s.isOpen(), is(true));
+                            Awaitility.await().atMost(Duration.ofSeconds(45)).until(() -> s.getServerKey() != null);
+                            assertThat(KnownHostsTestUtil.checkKeys(s), is(true));
+                            // Should have first connection and create a file
+                            assertThat(mockedKnownHosts, is(anExistingFile()));
+                            try {
+                                // Should connect and do not add new line because keys are equal
+                                assertThat(mockedKnownHosts, is(anExistingFile()));
+                                List<String> keys = Files.readAllLines(mockedKnownHosts.toPath());
+                                assertThat(keys.size(), is(1));
+                                assertThat(keys, is(Collections.singletonList(hostKeyEntry)));
+                            } catch (IOException e) {
+                                throw new RuntimeException(e);
+                            }
+                            return true;
+                        })
+                .close();
     }
 
     @Test
+    @Ignore("cannot verify a non hash host while we store hash host")
```

Comment:
test restored left over from initial work

Reply:
@MarkEWaite do yo have some logs? because it's passing locally

Reply:
I've uploaded a gist with the log that happens on my local Linux computer when I run:

```
  mvn clean -Dtest=AcceptFirstConnectionVerifierTest verify 2>&1 | tee logfile
```

I believe that the same message was reported on ci.jenkins.io , though a failing run is not available in the job history of that machine

Reply:
I've pushed 897001ec71e14808cb0e3bca73a2fa0eb450739d so that I can double check if the failure on ci.jenkins.io is the same as the failure that I see locally.  That will help me check if my local configuration is the cause of the local failure that I see.

Reply:
The [ci.jenkins.io **job](https://ci.jenkins.io/job/Plugins/job/git-client-plugin/view/change-requests/job/PR-1127/77/testReport/junit/org.jenkinsci.plugins.gitclient.verifier/AcceptFirstConnectionVerifierTest/linux_21___Build__linux_21____testVerifyServerHostKeyWhenHostnameWithoutPort/)** fails the test with the same message.  It may still be a configuration error, but if so, then the configuration error is both in my environment and on ci.jenkins.io